wasm/types.go

```diff
@@ -0,0 +1,234 @@
+package wasm
```

Comment:
I doubt that if we need this public `wasm`... at least looks almost everything is only used in internal pakages except interfaces `ModuleFunctions`, `HostFunctionCallContext` and `Memory`..

The reason for that is that maybe it seems annoying to have `internalwasm.Foo` all the time..

Reply:
it is difficult to enforce the naming `internalwasm` when there's not collision possibility, and I think there's no lint for such a thing

Reply:
actually I see `wasm` is used for internal/wasm in engine.go but `internalwasm` is used in jit_amd64.go..